### ISSUE TITLE
Error with setting value of n in ensemble mean calculation in distributed environment

### ISSUE THREAD

**Issue by ankurmahesh** [NVIDIA/modulus#57]
When I invoke the `__call__` method in Mean in `metrics/general/ensemble_metrics.py` in a distributed environment with modulus's distributed manager, I get the following error:

```
  File "/global/common/software/m1517/amahesh/fcn_mip-env/lib/python3.8/site-packages/modulus/metrics/general/ensemble_metrics.py", line 163, in __call__
    dist.all_reduce(self.n, op=dist.ReduceOp.SUM)
  File "/global/common/software/m1517/amahesh/fcn_mip-env/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 1451, in wrapper
    return func(*args, **kwargs)
  File "/global/common/software/m1517/amahesh/fcn_mip-env/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 1700, in all_reduce
    work = default_pg.allreduce([tensor], opts)
RuntimeError: Tensors must be CUDA and dense
```

This is because `n` gets set to the CPU in this [line](https://github.com/NVIDIA/modulus/blob/main/modulus/metrics/general/ensemble_metrics.py#L158).  (Even if `input` is a CUDA tensor, `torch.as_tensor(input.shape[0])` returns a torch.LongTensor, according to my tests.)

Perhaps the line above should be changed to 

`self.n = torch.as_tensor(input.shape[0]).to(self.device)`

For the time being, I fixed the issue by using Mean's `update` method instead of the `__call__` method, even for calculating the initial mean.  This change fixed the issue.

**Comment by akshaysubr** [comment]
Thanks for reporting this issue. Tagging @dallasfoster for further thoughts.

**Comment by dallasfoster** [comment]
Thanks for the issue submission. The intended behavior will be
```
self.n = torch.as_tensor([input.shape[batch_dim], device = self.device)
```
Note that it is preferred (although with minimal performance difference here) to initialize the tensor directly on the desired device.

Fixed by #63
